### PR TITLE
minor fixes based on IT feedback 

### PR DESCRIPTION
**Pull request by Abdiiir**
- Build timestamp for if clauses
- Avoid nullPointerException

### CODE REVIEW

File: jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void create(TimestampTO timestamp) {
       // Because the distinction is not visible after the timestamp has been created, so we cannot
       // rely on it
       // in general either way.
-      timestampTOBuilder = timestampTOBuilder.modeOfTransport(ModeOfTransport.VESSEL);
+      timestamp = timestampTOBuilder.modeOfTransport(ModeOfTransport.VESSEL).build();
```
Comment (nt-as) [inline jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java:60]: No, we are deliberately _not_ building here as there might be more changes.

File: jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public void create(TimestampTO timestamp) {
       // The errors for misaligned values are handled below.
       if (timestamp.facilitySMDGCode() == null &&
         locationTO.facilityCodeListProvider() == FacilityCodeListProvider.SMDG) {
-        timestampTOBuilder = timestampTOBuilder.facilitySMDGCode(locationTO.facilityCode());
+        timestamp = timestampTOBuilder.facilitySMDGCode(locationTO.facilityCode()).build();
```
Comment (nt-as) [inline jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java:93]: No, we are deliberately _not_ building here as there might be more changes.

**Review by FinnAtAsseco — APPROVED** [review]

**Comment by FinnAtAsseco** [comment]
Feels wasteful to build on every change, but I guess it's not harmful

**Review by nt-as — DISMISSED** [review]
See above

File: jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java [hunk 3]
```diff
@@ -59,7 +60,7 @@ public void create(TimestampTO timestamp) {
       // in general either way.
       timestampTOBuilder = timestampTOBuilder.modeOfTransport(ModeOfTransport.VESSEL);
     }
-    if (!timestamp.modeOfTransport().equals(ModeOfTransport.VESSEL)) {
+    else {
```
Comment (nt-as) [inline jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java:63]: I think this should have been an `else if`

**Review by nt-as — APPROVED** [review]

**Pull request merged**